Content.Server/Medical/CryoPodSystem.cs

```diff
@@ -0,0 +1,389 @@
+        SubscribeLocalEvent<CryoPodComponent, GetVerbsEvent<AlternativeVerb>>(AddAlternativeVerbs);
+        SubscribeLocalEvent<CryoPodComponent, DragDropEvent>(HandleDragDropOn);
+        SubscribeLocalEvent<CryoPodComponent, InteractHandEvent>(HandleInteractHand);
+        SubscribeLocalEvent<CryoPodComponent, InteractUsingEvent>(OnInteractUsing);
+        SubscribeLocalEvent<CryoPodComponent, DoInsertCryoPodEvent>(DoInsertCryoPod);
+        SubscribeLocalEvent<CryoPodComponent, ExaminedEvent>(OnExamined);
+        SubscribeLocalEvent<CryoPodComponent, PowerChangedEvent>(OnPowerChanged);
+        SubscribeLocalEvent<CryoPodComponent, GotEmaggedEvent>(OnEmagged);
+        SubscribeLocalEvent<CryoPodComponent, CryoPodPryFinished>(OnCryoPodPryFinished);
+        SubscribeLocalEvent<CryoPodComponent, CryoPodPryInterrupted>(OnCryoPodPryInterrupted);
+        SubscribeLocalEvent<CryoPodComponent, DestructionEventArgs>(OnDestroyed);
+        SubscribeLocalEvent<CryoPodComponent, GasAnalyzerScanEvent>(OnGasAnalyzed);
+
+    }
+
+    private void OnComponentInit(EntityUid uid, CryoPodComponent cryoPodComponent, ComponentInit args)
+    {
+        base.Initialize();
+        cryoPodComponent.BodyContainer = _containerSystem.EnsureContainer<ContainerSlot>(uid, $"scanner-bodyContainer");
+    }
+
+    private void UpdateAppearance(EntityUid uid, CryoPodComponent? cryoPod = null)
+    {
+        if (!Resolve(uid, ref cryoPod))
+            return;
+
+        var cryoPodEnabled = HasComp<ActiveCryoPodComponent>(uid);
+        _appearanceSystem.SetData(uid, SharedCryoPodComponent.CryoPodVisuals.IsOpen, cryoPod.BodyContainer.ContainedEntity == null);
+        _appearanceSystem.SetData(uid, SharedCryoPodComponent.CryoPodVisuals.IsOn, cryoPodEnabled);
+        if (TryComp<PointLightComponent>(uid, out var light))
+        {
+            light.Enabled = cryoPodEnabled && cryoPod.BodyContainer.ContainedEntity != null;
+        }
+
+        _appearanceSystem.SetData(uid,SharedCryoPodComponent.CryoPodVisuals.PanelOpen, false);
+    }
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+        foreach (var (_, cryoPod) in EntityQuery<ActiveCryoPodComponent, CryoPodComponent>())
+        {
+            cryoPod.Accumulator += frameTime;
+
+            if (cryoPod.Accumulator < cryoPod.BeakerTransferTime)
+                continue;
+
+            cryoPod.Accumulator -= cryoPod.BeakerTransferTime;
+
+            var container = _itemSlotsSystem.GetItem(cryoPod.Owner, "beakerSlot");
+            var patient = cryoPod.BodyContainer.ContainedEntity;
+            if (container != null
+                && container.Value.Valid
+                && patient != null
+                && _solutionContainerSystem.TryGetFitsInDispenser(container.Value, out var containerSolution))
+            {
+
+                var bloodstreamQuery = GetEntityQuery<BloodstreamComponent>();
```

Comment:
this doesnt actually change anything since you're not hoisting it out of a loop, just use a regular `TryComp`